api/repositories/droplet_repository.go

```diff
@@ -58,14 +59,21 @@ func (r DropletRecord) Relationships() map[string]string {
 }
 
 type ListDropletsMessage struct {
+	GUIDs        []string
 	PackageGUIDs []string
 	AppGUIDs     []string
+	SpaceGUIDs   []string
+}
+
+func (m *ListDropletsMessage) matches(droplet korifiv1alpha1.CFBuild) bool {
```

Comment:
We are currently pushing filtering to the server side leveraging kubernetes' label selectors. We no longer use the `matches` method for client side filtering.

The way this works is via a indexer webhook that puts various labels on each registered object that we can later lookup by. The right way to filter by guid would be to add a [rule](https://github.com/cloudfoundry/korifi/blob/386e2605057d5185dc580dcb59b7bcb6959d6853/controllers/webhooks/label_indexer/webhook.go#L44-L48) to the indexer webhook that puts the drioplet guid as a label in the CFBuild resource and then to translate message.GUIDs to `WithLabelIn(<guid-label-name>, m.GUIDs)` as it is done for space guids.